From 39adeffc9b69924c748a944f3a8d397e0477c26d Mon Sep 17 00:00:00 2001
From: Hanjun Guo <guohanjun@huawei.com>
Date: Wed, 23 Feb 2022 18:45:25 +0800
Subject: [PATCH] tty/amba-pl011: Call acpi_put_table() to fix memory leak
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 39adeffc9b69924c748a944f3a8d397e0477c26d
Modified-by-SEL: No


ascend inclusion
category: bugfix
bugzilla: https://gitee.com/src-openeuler/kernel/issues/I4U4KV
CVE: NA

----------------------------------------

acpi_get_table() should be coupled with acpi_put_table(), or
it will leat to memory leak, fix the memory leak to call
acpi_put_table().

Signed-off-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Lijun Fang <fanglijun3@huawei.com>
Reviewed-by: Weilong Chen <chenweilong@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/tty/serial/amba-pl011.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/tty/serial/amba-pl011.c b/drivers/tty/serial/amba-pl011.c
index 379d83432e7e..4f2c9378931e 100644
--- a/drivers/tty/serial/amba-pl011.c
+++ b/drivers/tty/serial/amba-pl011.c
@@ -1516,6 +1516,8 @@ static void pl011_check_hisi_workaround(void)
 			break;
 		}
 	}
+
+	acpi_put_table(tbl);
 }
 
 #else
-- 
2.26.2

